[nnc] Rename IRSimplifierBase with PolynomialBase

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62138
* #60693
* #60688
* #60687
* __->__ #60686

Differential Revision: [D29373316](https://our.internmc.facebook.com/intern/diff/D29373316)

huiguoo committed:
[nnc] Rename IRSimplifierBase with PolynomialBase

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29373316).

navahgar commented on torch/csrc/jit/tensorexpr/ir_simplifier.h:
```diff
@@ -404,9 +404,9 @@ class MinTerm : public ExprNode<MinTerm> {
 
 // Stmt simplification should occur in both modes.
-class TORCH_API IRSimplifierBase : public IRMutator {
+class TORCH_API PolynomialBase : public IRMutator {
```
What is the need for this change?

huiguoo replied:
The name indicates that it is the base class of all simplifications, but the context-sensitive simplification (this PR stack) is not based on it. It currently serves as the base class of `PolynomialTransformer` and `TermExpander` so I guess `PolynomialBase` is a more appropriate name?

huiguoo committed:
Update on "[nnc] Rename IRSimplifierBase with PolynomialBase"


Differential Revision: [D29373316](https://our.internmc.facebook.com/intern/diff/D29373316)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29373316).
